legacy/src/main/java/org/opensearch/sql/legacy/pit/PointInTimeHandlerImpl.java

```diff
@@ -37,47 +40,55 @@ public PointInTimeHandlerImpl(Client client, String[] indices) {
     this.indices = indices;
   }
 
-  /** Create PIT for given indices */
+  /**
+   * Constructor for class
+   *
+   * @param client OpenSearch client
+   * @param pitId Point In Time ID
+   */
+  public PointInTimeHandlerImpl(Client client, String pitId) {
+    this.client = client;
+    this.pitId = pitId;
+  }
+
+  /**
+   * Create PIT for given indices
+   *
+   * @return Point In Time creation status
+   */
   @Override
-  public void create() {
+  public boolean create() {
     CreatePitRequest createPitRequest =
         new CreatePitRequest(
             LocalClusterState.state().getSettingValue(SQL_CURSOR_KEEP_ALIVE), false, indices);
-    client.createPit(
-        createPitRequest,
-        new ActionListener<>() {
-          @Override
-          public void onResponse(CreatePitResponse createPitResponse) {
-            pitId = createPitResponse.getId();
-            LOG.info("Created Point In Time {} successfully.", pitId);
-          }
-
-          @Override
-          public void onFailure(Exception e) {
-            LOG.error("Error occurred while creating PIT", e);
-          }
-        });
+    ActionFuture<CreatePitResponse> execute =
+        client.execute(CreatePitAction.INSTANCE, createPitRequest);
+    try {
+      CreatePitResponse pitResponse = execute.get();
+      pitId = pitResponse.getId();
+      LOG.info("Created Point In Time {} successfully.", pitId);
+      return true;
+    } catch (InterruptedException | ExecutionException e) {
+      throw new RuntimeException("Error occurred while creating PIT.", e);
```

Comment:
What will be the end user response when we encounter these?

Reply:
User will see exception with the failure message followed by stack trace of underlying execution exception.

Reply:
Can you test it and see what is the error code and message that is coming up. You can add fake exception and test it.

Reply:
you can do this after merging the code as well.